scripts/download/tools.ts

```diff
@@ -0,0 +1,404 @@
+import { spawnSync } from 'child_process';
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+
+import {
+  download, downloadZip, downloadTarGZ, getResource, DownloadOptions, ArchiveDownloadOptions,
+} from '../lib/download';
+import DependencyVersions from './dependencies';
+
+type DependencyPlatform = 'wsl' | 'linux' | 'darwin' | 'win32';
+type Platform = 'linux' | 'darwin' | 'win32';
+type GoPlatform = 'linux' | 'darwin' | 'windows';
+
+type DownloadContext = {
+  versions: DependencyVersions;
+  dependencyPlaform: DependencyPlatform;
+  platform: Platform;
+  kubePlatform: GoPlatform;
```

Comment:
Similarly to the type, this should be `goPlatform` (or `goos`, naming it after the [environment variable](https://go.dev/doc/install/source#environment), in which case also rename the type).